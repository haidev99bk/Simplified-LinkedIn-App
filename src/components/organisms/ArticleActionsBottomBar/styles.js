import {StyleSheet} from 'react-native';
import color from '../../../styles/colors';

const styles = StyleSheet.create({
  container: {
    padding: 4,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  actionsInfo: {
    flexDirection: 'row',
    paddingVertical: 8,
    justifyContent: 'space-between',
  },
  likeInfoContainer: {
    flexDirection: 'row',
  },
  line: {
    height: 1,
    backgroundColor: color.darkGrey,
  },
  likeIcon: {
    height: 20,
    width: 20,
    marginRight: 8,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 12,
    backgroundColor: color.mainBlue,
  },
});

export default styles;
