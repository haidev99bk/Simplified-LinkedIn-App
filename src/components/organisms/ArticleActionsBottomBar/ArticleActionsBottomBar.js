import {View, Text} from 'react-native';
import React from 'react';
import styles from './styles';
import ArticleAction from '../../molecules/ArticleAction';
import Icon from '../../atoms/Icon';
import color from '../../../styles/colors';

export default function ArticleActionsBottomBar(props) {
  const {actions, comments, likes, hasLiked, articleId} = props;

  return (
    <View>
      <View style={styles.actionsInfo}>
        <View style={styles.likeInfoContainer}>
          <View style={styles.likeIcon}>
            <Icon family="AntDesign" name="like1" color={color.white} />
          </View>
          <Text style={styles.numberOfLikes}>{likes}</Text>
        </View>
        <View style={styles.commentInfoContainer}>
          <Text style={styles.numberOfLikes}>{comments + ' comments'}</Text>
        </View>
      </View>
      <View style={styles.line} />
      <View style={styles.actionsContainer}>
        <View style={styles.container}>
          {actions.map((action, index) => {
            let iconFamily = action.iconFamily;
            let iconName = action.iconName;
            let iconColor;

            if (action.label === 'Like' && hasLiked) {
              iconFamily = 'AntDesign';
              iconName = 'like1';
              iconColor = color.mainBlue;
            }
            return (
              <ArticleAction
                key={action.label + index}
                iconFamily={iconFamily}
                iconName={iconName}
                iconColor={iconColor && iconColor}
                label={action.label}
                onPress={() => {
                  action?.onPress && action.onPress(articleId);
                }}
              />
            );
          })}
        </View>
      </View>
    </View>
  );
}
