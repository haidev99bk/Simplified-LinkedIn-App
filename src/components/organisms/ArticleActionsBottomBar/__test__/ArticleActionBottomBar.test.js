import 'react-native';
jest.mock('react-native-device-info', () => {
  return {
    hasNotch: () => true,
  };
});
import React from 'react';
import {render} from '@testing-library/react-native';
import ArticleActionsBottomBar from '../ArticleActionsBottomBar';

const props = {
  actions: [
    {
      iconFamily: 'AntDesign',
      iconName: 'like2',
      label: 'Like',
    },
    {
      iconFamily: 'Foundation',
      iconName: 'loop',
      label: 'Repost',
    },
    {
      iconFamily: 'FontAwesome',
      iconName: 'comment',
      label: 'Comment',
    },
    {
      iconFamily: 'Feather',
      iconName: 'send',
      label: 'Send',
    },
  ],
  comments: 10,
  likes: 10,
  hasLiked: true,
  articleId: 2,
};

test('ArticleActionsBottomBar rendered correctly', () => {
  const snap = render(<ArticleActionsBottomBar {...props} />).toJSON();

  expect(snap).toMatchSnapshot();
});
