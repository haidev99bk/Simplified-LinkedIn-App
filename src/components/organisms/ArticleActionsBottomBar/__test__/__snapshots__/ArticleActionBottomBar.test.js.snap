// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowerArticles rendered correctly 1`] = `
<View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#1566c2",
            "borderRadius": 12,
            "height": 20,
            "justifyContent": "center",
            "marginRight": 8,
            "width": 20,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#fff",
                "fontSize": 12,
              },
              undefined,
              {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text>
        10
      </Text>
    </View>
    <View>
      <Text>
        10 comments
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#f3f2ee",
        "height": 1,
      }
    }
  />
  <View>
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 4,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "height": 24,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "height": 20,
                "justifyContent": "center",
                "width": 20,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#1566c2",
                    "fontSize": 16,
                  },
                  undefined,
                  {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            style={
              {
                "fontWeight": "600",
              }
            }
          >
            Like
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "height": 24,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "height": 20,
                "justifyContent": "center",
                "width": 20,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 16,
                  },
                  undefined,
                  {
                    "fontFamily": "fontcustom",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            style={
              {
                "fontWeight": "600",
              }
            }
          >
            Repost
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "height": 24,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "height": 20,
                "justifyContent": "center",
                "width": 20,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 16,
                  },
                  undefined,
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            style={
              {
                "fontWeight": "600",
              }
            }
          >
            Comment
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "height": 24,
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "height": 20,
                "justifyContent": "center",
                "width": 20,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": undefined,
                    "fontSize": 16,
                  },
                  undefined,
                  {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
          <Text
            style={
              {
                "fontWeight": "600",
              }
            }
          >
            Send
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
